ergo/platforms/predictit.py

```diff
@@ -0,0 +1,268 @@
+    """
+    A single binary question in a PredictIt market.
+
+    :param market: PredictIt question instance
+    :param data: Contract JSON retrieved from PredictIt API
+
+    :ivar PredictItQuestion market: predictit market instance
+    :ivar int id: id of the contract
+    :ivar datetime.datetime dateEnd: end-date of a market, usually None
+    :ivar str image: url of the image resource for the contract
+    :ivar str name: name of the contract
+    :ivar str shortName: shortened name of the contract
+    :ivar str status: status of the contract. Closed markets aren't included in the API, so always "Open"
+    :ivar float lastTradePrice: last price the contract was traded at
+    :ivar float bestBuyYesCost: cost to buy a single Yes share
+    :ivar float bestBuyNoCost: cost to buy a single No share
+    :ivar float bestSellYesCost: cost to sell a single Yes share
+    :ivar float bestSellNoCost: cost to sell a single No share
+    :ivar float lastClosePrice: price the contract closed at the previous day
+    :ivar int displayOrder: position of the contract in PredictIt. Defaults to 0 if sorted by lastTradePrice
+    """
+
+    def __init__(self, market: "PredictItMarket", data: Dict):
+        self.market = market
+        self._data = data
+
+    def __repr__(self):
+        return f'<PredictItQuestion title="{self.name}">'
+
+    def __getattr__(self, name: str):
+        """
+        If an attribute isn't directly on the class, check whether it's in the
+        raw contract data. If it's a time, format it appropriately.
+        :param name:
+        :return: attribute value
+        """
+        if name not in self._data:
+            raise AttributeError(
+                f"Attribute {name} is neither directly on this class nor in the raw question data"
+            )
+        if name != "dateEnd":
+            return self._data[name]
+        dateEnd = self._data["dateEnd"]
+        if dateEnd == "N/A":
+            return None
+        try:
+            return parse(dateEnd)
+        except ValueError:
+            print(f"The column {name} could not be converted into a datetime")
+            return dateEnd
+
+    def set_data(self, key: str, value: Any):
+        """
+        Set key on data dict
+        :param key:
+        :param value:
+        """
+        self._data[key] = value
+
+    @staticmethod
+    def to_dataframe(
+        questions: List["PredictItQuestion"], columns=None,
+    ) -> pd.DataFrame:
+        """
+        Summarize a list of questions in a dataframe
+        :param questions: questions to summarize
+        :param columns: list of column names as strings
+        :return: pandas dataframe summarizing the questions
+        """
+        if columns is None:
+            columns = ["id", "name", "dateEnd"]
+        data = [[question._data[key] for key in columns] for question in questions]
+
+        return pd.DataFrame(data, columns=columns)
+
+    def get_community_prediction(self) -> float:
```

Comment:
@stuhlmueller what do you think?

Reply:
Yeah, probably shouldn't try to standardize for now. I think it would be nice to have a uniform interface eventually but the current middle ground between standardized and custom isn't great. Better to start with platform-specific implementations and then consider whether and how to standardize later.